### PR TITLE
STYLE: Classes needing explicit instantiation have definitions in cxx files

### PR DESCRIPTION
**Pull request by hjmjohnson**
Move definition code from hxx to cxx file.
    
These classes *must* be explicitly
instantiated for each valid template type to support
dynamic casting robustly (one-definiton-rule).
    
As a consequence, the only place that the definitions
for these class member functions is needed is in the
cxx compiled files.

### CODE REVIEW

File: Modules/IO/TransformMatlab/include/itkMatlabTransformIO.hxx [hunk 1]
```diff
@@ -18,138 +18,5 @@
 #ifndef itkMatlabTransformIO_hxx
```
Comment (dzenanz) [inline Modules/IO/TransformMatlab/include/itkMatlabTransformIO.hxx]: Why is this file not completely removed?

Comment (hjmjohnson) [reply Modules/IO/TransformMatlab/include/itkMatlabTransformIO.hxx]: mistake.  It should be completely removed.

**Comment by blowekamp** [comment]
LGTM!

**Comment by hjmjohnson** [comment]
Committed in 0e5a2b1eb455a9772e2e4bc571bd0e53e19fc262

**Pull request closed without merging**